src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final String NAME_CLASS_BULLDOZER = "Bulldozer";
+    private static final String NAME_CLASS_EXCAVATOR = "Excavator";
+    private static final String NAME_CLASS_TRUCK = "Truck";
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
```

Comment:
```suggestion
    public List getAll(Class<? extends Machine> type) {
```
why list is not parametrized? what type should be used as a parameter?